File: EnigmaMachine/keyboard.java
package EnigmaMachine;

public class keyboard {
    String ASCII_UPPERCASE = "";

    public keyboard(){
        char c;
        for (c = 'A'; c <= 'Z'; c++) 
        {
            ASCII_UPPERCASE += c;
        }
    }

    public int forward(char letter){
        int signal = ASCII_UPPERCASE.indexOf(letter);
        return signal;

    }

    public char backward(int signal){
        char letter = ASCII_UPPERCASE.charAt(signal);
        return letter;

    }
}

File: EnigmaMachine/Python/Keyboard.py
from string import ascii_uppercase

class keyboard:                                      # This class has two properties
    def forward(self, letter):                       # To turn a letter into it's position in the 'ascii_uppercase'
        signal = ascii_uppercase.find(letter)
        return signal

    def backward(self, signal):                      # And turning a position from the 'ascii_uppercase' into a letter
        letter = ascii_uppercase[signal]
        return letter